### DISCUSSION TITLE
OAuth2 authentication with client_credentials

### DISCUSSION THREAD

**Discussion by bbenameur** [ThreeMammals/Ocelot#1877] [Show and tell]
## Expected Behavior / New Feature
I need to use an  OAuth2 Authentication, grant_type = client_credentials

my input are :   
1- Token url
2- ClientId
3- ClientSecret
4- Grant_Type : client_credentials

## Actual Behavior / Motivation for New Feature
I'm trying to use ocelot with OAuth2 Authentication, grant_type = client_credentials
but I have always a 401 http status code like a route is not authorized

## Steps to Reproduce the Problem
```csharp
    services
        .AddAuthentication()
        .AddOAuth2Introspection("TestToken", options =>
        {
            options.Authority = "https://xxxxxxxxxxxxxx/token";
            options.ClientId = "xxxxxxxxxxxxxxxxxx";
            options.ClientSecret = "xxxxxxxxxxxxxxxxxxxxx";
            options.AuthenticationType = "client_credentials";
        });

    var ocelotBuilder = services.AddOcelot().AddPolly();
```

on the **ocelot.json**: 
```json
  "AuthenticationOptions": {
        "AuthenticationProviderKey": "TestToken",
        "AllowedScopes": []
  },
```
even when i see the log there is not call to token url (https://xxxxxxxxxxxxxx/token)
## Specifications

  - Version:  Ocelot.Administration : 18.0.0/ OCelot 18.0.0
  - Platform: Windows / .Net 6
  - Subsystem:

**Comment by raman-m** [comment]
Hi Béchir!
Welcome to Ocelot world! 🐅 

Please show and attach all configuration artifacts when you connect to this downstream service directly, without Ocelot gateway.
So, prove that direct connection works.
After that we will start brainstorming how to setup Ocelot and the route.
